lime/text/UTFString.hx

```diff
@@ -0,0 +1,354 @@
+package lime.text;
+
+import haxe.Utf8;
+#if unifill
+import lime.text.unifill.Unifill;
+import lime.text.unifill.CodePoint;
+#end
+
+/**
+ * ...
+ * @author
+ */
+abstract UTFString(String) from String to String
+{
+	#if (unifill && (neko || php || cpp))
+	static var inited:Bool = false;
+	static var lcaseMap:Map<Int, Int>;
+	static var ucaseMap:Map<Int, Int>;
+	#end
+	
+	/**
+		The number of characters in `this` String.
+	**/
+	public var length(get, never) : Int;
+	
+	/**
+		Creates a copy from a given String.
+	**/
+	public function new(str:String)
+	{
+		this = new String(str);
+	}
+	
+	/**
+		Caching of character maps in two case sensitivites
+	**/
+	static function initialize() : Void {
+		#if (unifill && (neko || php || cpp))
+		lcaseMap = new Map<Int, Int>();
+		ucaseMap = new Map<Int, Int>();
+		
+		Utf8ExtInternal.fillUpperToLowerMap(lcaseMap);
+		Utf8ExtInternal.fillLowerToUpperMap(ucaseMap);
+		inited = true;
+		#end
+	}
+	
+	/**
+		Returns a String where all characters of `this` String are upper case.
+		
+		Affects the characters `a-z`. Other characters remain unchanged.
```

Comment:
Technically, this function is unicode aware and would affect more than just a-z ;)